client/packages/system/src/Patient/DetailView/DetailView.tsx

```diff
@@ -63,23 +73,90 @@ export const PatientDetailView: FC = () => {
   }, [patient]);
 
   const handleSave = useUpsertPatient();
-  const { JsonForm, isLoading } = useJsonForms(
+  const { JsonForm, saveData, revert, isSaving, isDirty } = useJsonForms(
     documentName,
     { handleSave },
     createDoc
   );
+  const save = useCallback(async () => {
+    const documentName = await saveData();
+    if (documentName) {
+      setDocumentName(documentName);
+      // patient has been created => unset the create request data
+      setNewPatient(undefined);
+    }
+  }, [saveData]);
 
   useEffect(() => {
     if (!documentName && currentPatient) {
       setDocumentName(currentPatient?.document?.name);
     }
   }, [currentPatient]);
 
-  if (isLoading) return <DetailViewSkeleton />;
+  const showSaveConfirmation = useConfirmationModal({
+    onConfirm: save,
+    message: t('messages.confirm-save-generic'),
+    title: t('heading.are-you-sure'),
+  });
+
+  const showCancelConfirmation = useConfirmationModal({
+    onConfirm: () => {
+      if (createDoc) {
+        setNewPatient(undefined);
+        navigate(
+          RouteBuilder.create(AppRoute.Dispensary)
+            .addPart(AppRoute.Patients)
+            .build()
+        );
+      } else {
+        revert();
+      }
+    },
+    message: t('messages.confirm-cancel-generic'),
+    title: t('heading.are-you-sure'),
+  });
+
+  return (
+    <>
+      <Box style={{ position: 'absolute', bottom: 40, right: 30, zIndex: 100 }}>
+        <Box gap={0.5} flexDirection="row" display="flex" alignItems="center">
+          {createDoc ? (
+            <DialogButton
+              variant="cancel"
+              disabled={!isDirty || isSaving}
+              onClick={() => {
+                showCancelConfirmation();
+              }}
+            />
+          ) : (
+            <LoadingButton
+              color="secondary"
+              disabled={!isDirty || isSaving}
+              isLoading={false}
+              onClick={() => showCancelConfirmation()}
+            >
+              {t('button.revert')}
+            </LoadingButton>
+          )}
+          <LoadingButton
+            color="secondary"
+            disabled={!isDirty}
+            isLoading={isSaving}
+            onClick={() => showSaveConfirmation()}
+          >
+            {createDoc ? t('button.create') : t('button.save')}
+          </LoadingButton>
+        </Box>
+      </Box>
+      {JsonForm}
+    </>
+  );
+};
 
+export const PatientView: FC = () => {
```

Comment:
the current pattern is to have the file name match the exported component; in this case I would name the component `DetailView`

Reply:
thanks, the PatientView is the whole patient view, i.e. summary + tabs. The DetailView is the details tabs. For this reason I renamed the file/dir to PatientView